src/odemis/util/comp.py

```diff
@@ -108,3 +110,19 @@ def get_fov_rect(comp, fov):
             center[1] - fov[1] / 2,  # top
             center[0] + fov[0] / 2,  # right
             center[1] + fov[1] / 2)  # bottom
+
+def execute_function_within_subfuture(future, fn, args):
```

Comment:
Let's not put it in util.comp, yet. It relies on a lot of non-standard behaviour of the futures. Especially, the moving_lock and _must_stop are special attributes placed by the actuator code. 
So I'd suggest to place this function somewhere inside the actuator.LinkedHeightActuator.

That said, in the long term, I agree that it'd make sense to share this code as we are starting to have many places in Odemis which do the same. Maybe it would be possible to extend the `odemis.model._future.CancellableFuture` class to provide such mechanism. In such case, it'd handle the cancellation, and also the lock, and cancelling the sub futures.